airflow/operators/python_operator.py

```diff
@@ -167,21 +167,28 @@ def execute(self, context):
         logging.info('Skipping downstream tasks...')
         session = settings.Session()
 
+        downstream_tasks = context['task'].get_flat_relatives(upstream=False)
+        downstream_task_ids = [d.task_id for d in downstream_tasks]
+
+        logging.debug("Downstream task_ids {}".format(downstream_task_ids))
+
         TI = TaskInstance
         tis = session.query(TI).filter(
             TI.execution_date == context['ti'].execution_date,
-            TI.task_id.in_(context['task'].downstream_task_ids),
+            TI.task_id.in_(downstream_task_ids),
```

Comment:
Yeah, you're right, there's not a better function, but maybe we should add one. We should try to reduce SQL queries outside of models, especially since this one is incorrect. (task names are not unique globally)